runner/controller.go

```diff
@@ -0,0 +1,16 @@
+package runner
+
+type Controller interface {
+	// Run prepares cmd. It returns its status and any errors.
+	// Run may:
+	// enqueue cmd (leading to state PENDING)
+	// run cmd immediately (leading to state RUNNING)
+	// check if cmd is well-formed, and reject it if not (leading to state FAILED)
+	// wait a very short period of time for cmd to start (leading to any state)
+	// Run may not wait indefinitely for cmd to finish. This is an async API.
+	Run(cmd *Command) (RunStatus, error)
```

Comment:
based on prior conversation with Dan.... should we just return RunStatus (eliminate error) - put errors from the scoot platform in the RunStatus's .error field?  This applies to all APIs that return a status and an error value